Doc/c-api/intro.rst

```diff
@@ -174,6 +174,17 @@ complete listing.
    .. versionchanged:: 3.8
       MSVC support was added.
 
+.. c:macro:: PyDoc_STRVAR(name, str)
+
+   A shortcut for ``static const char name[] = str``, commonly used
```

Comment:
I don't think mentioning what this does internally is very useful, makes it somewhat contractual and could get out of date.

Maybe reword it just to be something along the lines of:

> Creates an object with name `name` that can be used in module, function, classes etc docstrings.

Reply:
Agreed, that's also why we typically avoid listing every possible use case, to avoid creating unnecessary contractual obligations. It adds to the number of potential backwards compatibility concerns.